### PR TITLE
Rename asObservable() to obs() (primitive types only)

### PR DESCRIPTION
**Pull request by subzero911**
*`Fixes 2.0.7`

I want to push a little fix for my 2.0.7 update. It was very recent, and this fix won't touch many users.
I found in practice that `1.asObservable()` is ugly and has no big difference to `ObservableInt(1)`
So I changed it to `.obs()`
It is short and convenient for GetX users (which use `.obs` getter) to jump to MobX.
```dart
final x = 1.obs();
```

### CODE REVIEW

**Comment by subzero911** [comment]
@pavanpodila can you approve the MR?

**Comment by pavanpodila** [comment]
> @pavanpodila can you approve the MR?

there seems to be a big drop in coverage. Can you please look into that @subzero911  ?

**Comment by subzero911** [comment]
`codecov/patch`
https://github.com/mobxjs/mobx.dart/pull/831/checks?check_run_id=6981802810
The check fails on lines 5, 12, 26, 33
These are function signatures - how am I supposed to test them? It seems to be impossible
Tests for function bodies are written.

Anyway, I added some tests for typedefs (to ensure they are corresponding to their generic types).

`codecov/project/flutter_mobx — 96.90% (-3.10%) compared to 4f98362`
I hadn't touch the `flutter_mobx` project - how could it drop the coverage?

File: mobx/lib/src/core/typedefs.dart [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
 
 /// aliases for primitive types
-typedef ObservableInt = Observable<int>;
-typedef ObservableDouble = Observable<double>;
-typedef ObservableBool = Observable<bool>;
-typedef ObservableString = Observable<String>;
+typedef ObsInt = Observable<int>;
+typedef ObsDouble = Observable<double>;
+typedef ObsBool = Observable<bool>;
+typedef ObsString = Observable<String>;
```
Comment (pavanpodila) [inline mobx/lib/src/core/typedefs.dart:7]: I don't see a need for these typedefs...Its better to expose the raw generic type directly...more readable that way

**Review by pavanpodila — CHANGES_REQUESTED** [review]

File: mobx/lib/src/core/typedefs.dart (hunk 1 above)
Comment (subzero911) [reply mobx/lib/src/core/typedefs.dart:7]: I agree, i will remove these typedefs.
They just make a code more tangled.

Comment (subzero911) [reply mobx/lib/src/core/typedefs.dart]: @pavanpodila done.

Comment (gabrielscr) [reply mobx/lib/src/core/typedefs.dart]: @pavanpodila please fix Observer, it's not reacting to any observables

Comment (subzero911) [reply mobx/lib/src/core/typedefs.dart]: @gabrielscr how does it relate to my MR?
Create your own issue and ask for help there.

Comment (gabrielscr) [reply mobx/lib/src/core/typedefs.dart]: @subzero911 no

**Review by pavanpodila — APPROVED** [review]

**Pull request merged**